src/background/controller/serviceDefinition.ts

```diff
@@ -51,16 +57,83 @@ export function serviceDefinition(address, keyId, type, network, opts = {}) {
     definition.method = 'EXT/RPC';
     definition.data = opts;
   }
+  return definition;
+}
+
+export async function httpPayerServiceDefinition(address, keyId, type, network, opts = {}) {
+  const app = initializeApp(getFirbaseConfig(), process.env.NODE_ENV);
+
+  const idToken = await getAuth(app).currentUser.getIdToken();
+  const definition = {
+    f_type: 'Service',
+    f_vsn: '1.0.0',
+    type: type,
+    uid: `fcw#${type}`,
+    method: 'HTTP/POST',
+    network: network || 'unknown',
+    endpoint: `${functionsUrl}/payer`,
+    identity: {
+      address: address,
+      keyId: keyId,
+    },
+    params:
+      opts && opts.params
+        ? opts.params
+        : {
+            idToken,
+          },
+  };
+
+  return definition;
+}
+
+export async function httpProposerServiceDefinition(address, keyId, type, network, opts = {}) {
+  const app = initializeApp(getFirbaseConfig(), process.env.NODE_ENV);
+
+  const idToken = await getAuth(app).currentUser.getIdToken();
+  const definition = {
+    f_type: 'Service',
+    f_vsn: '1.0.0',
+    type: type,
+    uid: `fcw#${type}`,
+    method: 'HTTP/POST',
+    network: network || 'unknown',
+    endpoint: `${functionsUrl}/proposer`,
+    identity: {
+      address: address,
+      keyId: keyId,
+    },
+    params:
+      opts && opts.params
+        ? opts.params
+        : {
+            idToken,
```

Comment:
The main concern I have is here, it seem like the dApp are able to get idToken and pass all our backend check.